include/fmt/std.h

```diff
@@ -36,6 +36,9 @@
 #  if FMT_HAS_INCLUDE(<optional>)
 #    include <optional>
 #  endif
+#  if FMT_HAS_INCLUDE(<expected>)
+#    include <expected>
+#  endif
 #endif
```

Comment:
Move it outside ``#if FMT_CPLUSPLUS >= 201703L``.

And use C++23 condition to avoid msvc warning:
```c++
#if FMT_HAS_INCLUDE(<expected>) && FMT_CPLUSPLUS > 202002L
#  include <expected>
#endif
```